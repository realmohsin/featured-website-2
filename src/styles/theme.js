import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
