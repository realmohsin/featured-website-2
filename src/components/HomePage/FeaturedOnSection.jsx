import React from 'react'
import clsx from 'clsx'
import { graphql, useStaticQuery } from 'gatsby'
import { GatsbyImage, getImage } from 'gatsby-plugin-image'
import { makeStyles } from '@material-ui/core/styles'
import { Container } from '@material-ui/core'

const useStyles = makeStyles(theme => ({
  featuredOnSection: {
    padding: '12rem 0',
    backgroundColor: 'white'
  },
  featuredOnTitle: {
    ...theme.custom.title3,
    textAlign: 'center',
    fontSize: '2rem',
    marginBottom: '3.5rem',
    textTransform: 'uppercase',
    letterSpacing: '0.5px'
    // ...theme.custom.outlineTitle
  },
  featuredGrid: {
    display: 'grid',
    gridTemplateColumns: 'repeat(5, 1fr)',
    rowGap: '3rem',
    [theme.breakpoints.down('md')]: {
      gridTemplateColumns: 'repeat(4, 1fr)'
    },
    [theme.breakpoints.down('sm')]: {
      gridTemplateColumns: 'repeat(3, 1fr)'
    },
    [theme.breakpoints.down('xs')]: {
      gridTemplateColumns: 'repeat(2, 1fr)'
    }
  },
  gridItem: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  featuredCompanyImg: {
    width: '19.1rem'
  },
  divider: {
    width: '100%',
    height: '2px',
    backgroundColor: theme.palette.secondary.main,
    marginBottom: '2rem'
  },
  bottomDivider: {
    marginTop: '8rem'
  },
  recognitionGrid: {
    display: 'grid',
    gridTemplateColumns: 'repeat(4, 1fr)',
    rowGap: '3rem',
    [theme.breakpoints.down('sm')]: {
      gridTemplateColumns: 'repeat(3, 1fr)'
    },
    [theme.breakpoints.down('xs')]: {
      gridTemplateColumns: 'repeat(2, 1fr)'
    }
  },
  superLawyersImg: {
    width: '19.1rem'
  }
}))

const FeaturedOnSection = () => {
  const classes = useStyles()
  const data = useStaticQuery(graphql`
    query {
      featured: allFile(filter: { name: { regex: "/^featured-/" } }) {
        nodes {
          childImageSharp {
            gatsbyImageData
          }
        }
      }
      recognition: allFile(filter: { name: { regex: "/^recognition-/" } }) {
        nodes {
          childImageSharp {
            gatsbyImageData
          }
        }
      }
    }
  `)

  console.log(data)

  return (
    <section className={classes.featuredOnSection}>
      <Container>
        <div className={classes.divider} />
        <h2 className={classes.featuredOnTitle}>Featured On</h2>
        <div className={classes.featuredGrid}>
          {data.featured.nodes.map((node, i) => (
            <div className={classes.gridItem}>
              <GatsbyImage
                image={getImage(node.childImageSharp)}
                alt='Featured In'
                className={classes.featuredCompanyImg}
              />
            </div>
          ))}
        </div>
        <div className={clsx(classes.divider, classes.bottomDivider)} />
        <h2 className={classes.featuredOnTitle}>Recognition</h2>
        <div className={classes.recognitionGrid}>
          {data.recognition.nodes.map((node, i) => (
            <div className={classes.gridItem}>
              <GatsbyImage
                image={getImage(node.childImageSharp)}
                alt='Recognition Award'
                className={classes.featuredCompanyImg}
              />
            </div>
          ))}
        </div>
      </Container>
    </section>
  )
}

export default FeaturedOnSection
