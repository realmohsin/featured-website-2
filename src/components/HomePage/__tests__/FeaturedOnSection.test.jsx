import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/core/styles';
import { createMuiTheme } from '@material-ui/core/styles';
import { useStaticQuery } from 'gatsby';
import FeaturedOnSection from '../FeaturedOnSection';

// Mock Gatsby Image
jest.mock('gatsby-plugin-image', () => {
  const React = require('react');
  const MockImage = (props) => (
    <img {...props} alt={props.alt} data-testid="gatsby-image" />
  );
  return {
    __esModule: true,
    GatsbyImage: MockImage,
    getImage: () => ({
      layout: 'constrained',
      width: 100,
      height: 50,
      images: {
        fallback: {
          src: 'test-image.jpg',
          srcSet: 'test-image.jpg 1x',
        },
      },
    }),
  };
});

// Mock theme
const theme = createMuiTheme({
  custom: {
    title3: {
      fontSize: '2rem',
      lineHeight: 1.2,
      fontWeight: 600,
    },
  },
  palette: {
    secondary: {
      main: '#000000',
    },
  },
  breakpoints: {
    down: () => false,
  },
});

describe('FeaturedOnSection', () => {
  const mockData = {
    featured: {
      nodes: [
        { childImageSharp: { gatsbyImageData: {} } },
        { childImageSharp: { gatsbyImageData: {} } },
        { childImageSharp: { gatsbyImageData: {} } },
      ],
    },
    recognition: {
      nodes: [
        { childImageSharp: { gatsbyImageData: {} } },
        { childImageSharp: { gatsbyImageData: {} } },
      ],
    },
  };

  beforeEach(() => {
    useStaticQuery.mockImplementation(() => mockData);
  });

  const renderWithTheme = (component) => {
    return render(
      <ThemeProvider theme={theme}>
        {component}
      </ThemeProvider>
    );
  };

  it('renders the featured on section with title', () => {
    renderWithTheme(<FeaturedOnSection />);
    
    // Check if the section is rendered
    const section = screen.getByRole('heading', { name: /featured on/i }).closest('section');
    expect(section).toBeInTheDocument();
    
    // Check if the title is rendered
    const title = screen.getByRole('heading', { name: /featured on/i, level: 2 });
    expect(title).toBeInTheDocument();
  });

  it('renders featured company images', () => {
    renderWithTheme(<FeaturedOnSection />);
    
    // Check if featured images are rendered
    const featuredImages = screen.getAllByAltText('Featured In');
    expect(featuredImages.length).toBe(mockData.featured.nodes.length);
    
    // Check if images are rendered with the correct alt text
    featuredImages.forEach((img) => {
      expect(img).toHaveAttribute('alt', 'Featured In');
    });
  });

  it('renders recognition section with title', () => {
    renderWithTheme(<FeaturedOnSection />);
    
    // Check if the recognition title is rendered
    const recognitionTitle = screen.getByRole('heading', { name: /recognition/i, level: 2 });
    expect(recognitionTitle).toBeInTheDocument();
    
    // Check if recognition images are rendered
    const recognitionImages = screen.getAllByAltText('Recognition Award');
    expect(recognitionImages.length).toBe(mockData.recognition.nodes.length);
  });

  it('matches snapshot', () => {
    const { container } = renderWithTheme(<FeaturedOnSection />);
    expect(container).toMatchSnapshot();
  });
});
