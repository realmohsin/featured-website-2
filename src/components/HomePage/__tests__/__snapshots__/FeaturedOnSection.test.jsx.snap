// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedOnSection matches snapshot 1`] = `
<div>
  <section
    class="makeStyles-featuredOnSection-28"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="makeStyles-divider-33"
      />
      <h2
        class="makeStyles-featuredOnTitle-29"
      >
        Featured On
      </h2>
      <div
        class="makeStyles-featuredGrid-30"
      >
        <div
          class="makeStyles-gridItem-31"
        >
          <img
            alt="Featured In"
            class="makeStyles-featuredCompanyImg-32"
            data-testid="gatsby-image"
            image="[object Object]"
          />
        </div>
        <div
          class="makeStyles-gridItem-31"
        >
          <img
            alt="Featured In"
            class="makeStyles-featuredCompanyImg-32"
            data-testid="gatsby-image"
            image="[object Object]"
          />
        </div>
        <div
          class="makeStyles-gridItem-31"
        >
          <img
            alt="Featured In"
            class="makeStyles-featuredCompanyImg-32"
            data-testid="gatsby-image"
            image="[object Object]"
          />
        </div>
      </div>
      <div
        class="makeStyles-divider-33 makeStyles-bottomDivider-34"
      />
      <h2
        class="makeStyles-featuredOnTitle-29"
      >
        Recognition
      </h2>
      <div
        class="makeStyles-recognitionGrid-35"
      >
        <div
          class="makeStyles-gridItem-31"
        >
          <img
            alt="Recognition Award"
            class="makeStyles-featuredCompanyImg-32"
            data-testid="gatsby-image"
            image="[object Object]"
          />
        </div>
        <div
          class="makeStyles-gridItem-31"
        >
          <img
            alt="Recognition Award"
            class="makeStyles-featuredCompanyImg-32"
            data-testid="gatsby-image"
            image="[object Object]"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
